src/rgw/rgw_auth.cc

```diff
@@ -68,3 +72,153 @@ rgw_auth_transform_old_authinfo(req_state * const s)
    * through any security check. */
                                     s->system_request));
 }
+
+
+/* RGWRemoteAuthApplier */
+int RGWRemoteAuthApplier::get_perms_from_aclspec(const aclspec_t& aclspec) const
+{
+  const auto iter = aclspec.find(info.auth_user.to_str());
+  if (std::end(aclspec) == iter) {
+    return iter->second;
+  }
+
+  return 0;
+}
+
+bool RGWRemoteAuthApplier::is_admin_of(const rgw_user& uid) const
+{
+  return info.is_admin;
+}
+
+bool RGWRemoteAuthApplier::is_owner_of(const rgw_user& uid) const
+{
+  if (info.acct_user.tenant.empty()) {
+    const rgw_user tenanted_acct_user(info.acct_user.id, info.acct_user.id);
+
+    if (tenanted_acct_user == uid) {
+      return true;
+    }
+  }
+
+  return info.acct_user == uid;
+}
+
+void RGWRemoteAuthApplier::create_account(const rgw_user acct_user,
+                                          RGWUserInfo& user_info) const      /* out */
+{
+  rgw_user new_acct_user = acct_user;
+
+  /* Administrator may enforce creating new accounts within their own tenants.
+   * The config parameter name is kept due to legacy. */
+  if (new_acct_user.tenant.empty() && g_conf->rgw_keystone_implicit_tenants) {
+    new_acct_user.tenant = new_acct_user.id;
+  }
+
+  user_info.user_id = new_acct_user;
+  user_info.display_name = info.acct_name;
+
+  int ret = rgw_store_user_info(store, user_info, nullptr, nullptr,
+                                real_time(), true);
+  if (ret < 0) {
+    ldout(cct, 0) << "ERROR: failed to store new user info: user="
+                  << user_info.user_id << " ret=" << ret << dendl;
+    throw ret;
+  }
+}
+
+/* TODO(rzarzynski): we need to handle display_name changes. */
```

Comment:
We need to make some kind of synchronization mechanism to update our local copy of Keystone-dependent pieces of information like the `display_name`. We never had such machinery. As a consequence, we might use an outdated tenant name (name, not UUID) that was stored in `RGWUserInfo` during  RadosGW's user creation.